src/main/java/org/logstash/input/SinceDB.java

```diff
@@ -0,0 +1,124 @@
+package org.logstash.input;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.logstash.common.io.DeadLetterQueueReader;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+class SinceDB {
+
+    final static class UnassignedSinceDB extends SinceDB {
+
+        private UnassignedSinceDB(Path sinceDbPath) {
+            super(sinceDbPath, Paths.get(System.getProperty("user.home")), 0);
+        }
```

Comment:
Is there a reason you're setting the current segment to `user.home`? This doesn't seem like it can ever be the case?
Or is this to ensure that `retrieveNewPosition` can work as a method to construct a `SinceDB` object by using its local copy of `sinceDbPath`?

Reply:
I used `user.home` as a no-op value, could it be a `/tmp` also. It's only used so that the object constructed and is able to create the new instance in  `retrieveNewPosition`